sdk/web-pubsub/web-pubsub-express/src/cloudEventsDispatcher.ts

```diff
@@ -52,6 +59,15 @@ function getConnectResponseHandler(
       response.statusCode = code;
       response.end(detail ?? "");
     },
+    failWith(res: ConnectEventErrorResponse | MqttConnectEventErrorResponse) {
+      if ("mqtt" in res) {
+        response.statusCode = getStatusCodeFromMqttConnectCode(res.mqtt.code);
```

Comment:
check `if(res.mqtt.code)` exists? how about MqttConnectEventErrorResponse to extend ConnectEventErrorResponse and `response.statusCode =  res.code ??getStatusCodeFromMqttConnectCode(res?.mqtt?.code)`

Reply:
This will be an issue when user defines both the `res.code` and `res.mqtt.code` in their response. So in this case, which code should we prioritize? Should we use the `res.code` or the code derived from `getStatusCodeFromMqttConnectCode(res?.mqtt?.code)`?

Reply:
If we make these two interfaces exclusive (no extension), this case won't happen. And I think that makes more sense since the usage for both cases are different. For current webpubsub, we return the status code directly from the response that user defined. For MQTT, we map the MQTT code user defined to HTTP status code, and return it

Reply:
sounds cool, strictly follow the spec

Reply:
Then what if user calls fail(code, detail) to fail a mqtt client? Sounds also allowed. Maybe we always use the outer side `code` as the code we use, if not set, we calculate from the inner mqtt code.

Reply:
I don't think we can use `fail(code,detail)` to fail a mqtt client since we need to match the [spec](https://learn.microsoft.com/en-us/azure/azure-web-pubsub/reference-mqtt-cloud-events#error-response-format), which requires an `MQTT return code`. As your suggestion, do you mean the code in the `fail(code,detail)` would  be considered as the `MQTT return code`?

Reply:
This also depends on whether or not we let the user define the HTTP Status Code for MQTT Error Response. If we strictly need to follow the mapping between MQTT Return Code and HTTP Status Code, then we need to separate these two. Otherwise we can do as what you suggested and combine these two interfaces

Reply:
@vicancy so just had some discussion with @Y-Sindo about this. Seems like we don't need to follow strictly with the mapping, so we can support `failWith({ code, detail })` for MQTT by randomly assigning a MQTT return code based on the `code` param (e.g 401 would be assigned with `MqttV311ConnectReturnCode.NotAuthorized`). And user can either define `failWith({ code, detail })`, or `failWith({ mqtt: { code, reason } })` for MQTT case. Do we still need to extend the interface for this case then?

Reply:
> randomly assigning a MQTT return code based on the code param (e.g 401 would be assigned with MqttV311ConnectReturnCode.NotAuthorized).

Our runtime is doing this?

> Do we still need to extend the interface in this case then?

Get your point, yes no need to. 

Another question: shall `failWith({code, detail})` keep consistent with `fail(code, detail)` to return detail as the response body?